### PR TITLE
122 bid won event

### PR DESCRIPTION
**Pull request by jeteve**
This is the guAnalytics side of https://trello.com/c/d6JaB1Jh

It sends the BidWon event (so when the winning bid effectively fills the Ad slot instead of Google) on its own to be later processed by the Prebid ETL job(s).

### CODE REVIEW

File: test/spec/modules/guAnalyticsAdapter_spec.js [hunk 1]
```diff
@@ -153,12 +158,19 @@ describe('Gu analytics adapter', () => {
...
     ev = JSON.parse(ajaxStub.firstCall.args[2]).hb_ev;
     expect(ev[4]).to.be.eql({ev: 'end', aid: '5018eb39-f900-4370-b71e-3bb5b48d324f', ttr: 447});
   });
+
+  it('should handle bid won event', () => {
+      events.emit(CONSTANTS.EVENTS.BID_WON, BIDONE);
+      let ev = analyticsAdapter.context.queue.peekAll();
+      expect(ev).to.have.length(0); // The queue has been flushed.
+      ev = JSON.parse(ajaxStub.firstCall.args[2]).hb_ev;
+      expect(ev[0]).to.be.eql({ev: 'bidwon', aid: '5018eb39-f900-4370-b71e-3bb5b48d324f', bid: BIDONE.bidId , sid: BIDONE.adUnitCode });
```
Comment (kelvin-chappell) [inline test/spec/modules/guAnalyticsAdapter_spec.js]: Do we need the ad unit code?

File: test/spec/modules/guAnalyticsAdapter_spec.js [hunk 2]
```diff
@@ -23,7 +20,13 @@ describe('Gu analytics adapter', () => {
       bidId: '208750227436c1',
       bidderRequestId: '1a6fc81528d0f7',
       auctionId: '5018eb39-f900-4370-b71e-3bb5b48d324f'
-    }],
+  };
+
+  const REQUEST1 = {
```
Comment (kelvin-chappell) [inline test/spec/modules/guAnalyticsAdapter_spec.js]: Not sure why these are in uppercase.  Perhaps I copied it from somewhere.

File: modules/guAnalyticsAdapter.js [hunk 3]
```diff
@@ -31,13 +36,16 @@ let analyticsAdapter = Object.assign(adapter({analyticsType}),
...
+          handler = trackBidWon;
+          break;
       }
       if (handler) {
         let events = handler(args);
         if (events && analyticsAdapter.context.queue) {
           analyticsAdapter.context.queue.push(events);
         }
-        if (eventType === CONSTANTS.EVENTS.AUCTION_END) {
+        if ( SENDALL_ON[eventType] ) {
```
Comment (kelvin-chappell) [inline modules/guAnalyticsAdapter.js]: Nice optimisation.

File: modules/guAnalyticsAdapter.js [hunk 4]
```diff
@@ -67,6 +75,17 @@ function sendAll() {
...
 }
 
+function trackBidWon(args){
+    const event = createHbEvent( args.bidderCode, 'bidwon',
+                                 args.adUnitCode,
+                                 args.auctionId,
+                                 undefined, // timeToRespond
+                                 undefined, // startTime
+                                 args.bidId
+                               );
```
Comment (kelvin-chappell) [inline modules/guAnalyticsAdapter.js]: I think we only need 'bidWon', auctionId and bidId.

Comment (jeteve) [reply modules/guAnalyticsAdapter.js]: Yep. I will remove all the other stuff :)

**Pull request merged**